src/nvim/api/vim.c

```diff
@@ -1562,3 +1563,15 @@ Object nvim_get_proc(Integer pid, Error *err)
 #endif
   return rvobj;
 }
+
+Integer nvim_get_caps_state(Error *err )
```

Comment:
I do not see why you should use `Integer` here. With `Error *err` argument you do have at least three states with mere `Boolean`: true, false and various errors.

Reply:
I had a thought, that if there was error then I would return -1, and 0 for no and 1 for yes, Boolean was option, I think of it

Reply:
You do not return anything at all with a error. You may see how dispatch coded, it generates code like

```C
Object handle_nvim_get_current_line(uint64_t channel_id, Array args, Error *error)
{
  Object ret = NIL;

  if (args.size != 0) {
    api_set_error(error, kErrorTypeException, "Wrong number of arguments: expecting 0 but got %zu", args.size);
    goto cleanup;
  }

  String rv = nvim_get_current_line(error);

  if (ERROR_SET(error)) {
    goto cleanup;
  }

  ret = STRING_OBJ(rv);

cleanup:
  return ret;
}
```

where you may clearly see NIL returned in case of error. Lua

```
static int nlua_msgpack_nvim_get_current_line(lua_State *lstate)
{
  Error err = ERROR_INIT;
  if (lua_gettop(lstate) != 0) {
    api_set_error(&err, kErrorTypeValidation, "Expected 0 arguments");
    goto exit_0;
  }
  const String ret = nvim_get_current_line(&err);
  nlua_push_String(lstate, ret);
  api_free_string(ret);


exit_0:
  if (ERROR_SET(&err)) {
    luaL_where(lstate, 1);
    lua_pushstring(lstate, err.msg);
    api_clear_error(&err);
    lua_concat(lstate, 2);
    return lua_error(lstate);
  }

  return 1;
}
```

may seem different as you could not actually see it not returning anything: value is pushed onto stack regardless of error condition (AFAIR in order to not rely on function not returning something allocated just in case). But `lua_error` is a longjump and the target may and will expect return value from another function, which may be the caller of the `nlua_msgpack_nvim_get_current_line` and not API function itself, with caller having a different contract, or an error condition (what I am saying is that lua error handling with `pcall` and longjumps does not allow failing functions to have meaningful return value even if authors were willing to implement it, this is not the case with VimL where builtin functions do return in case of failure, though you need to be careful not to put failing function inside `try` or do some hacking with `silent` to actually utilize that).

Reply:
(Dispatch is in `build/src/nvim/auto/api/private/dispatch_wrappers.generated.h`, lua wrappers are in `build/src/nvim/auto/msgpack_lua_c_bindings.generated.c`, actual generators are both in `src/nvim/generators/gen_api_dispatch.lua`, `src/nvim/CMakeLists.txt` may be used to connect dots.)